component/src/main/java/org/wso2/extension/siddhi/execution/env/ResourceBatchWindowProcessor.java

```diff
@@ -0,0 +1,279 @@
+                }
+            } else {
+                throw new SiddhiAppValidationException("Resource Batch window should have constant "
+                        + "for resource name parameter but found a dynamic attribute " + attributeExpressionExecutors[1]
+                        .getClass().getCanonicalName());
+            }
+            this.groupKeyExpressionExecutor = attributeExpressionExecutors[1];
+            if (attributeExpressionExecutors.length == 3) {
+                this.groupKeyExpressionExecutor = attributeExpressionExecutors[1];
+                if (attributeExpressionExecutors[2].getReturnType() == Attribute.Type.INT) {
+                    timeInMilliSeconds = Integer.parseInt(String.valueOf(((ConstantExpressionExecutor)
+                            attributeExpressionExecutors[2]).getValue()));
+                } else if (attributeExpressionExecutors[2].getReturnType() == Attribute.Type.LONG) {
+                    timeInMilliSeconds = Long.parseLong(String.valueOf(((ConstantExpressionExecutor)
+                            attributeExpressionExecutors[2]).getValue()));
+                } else {
+                    throw new SiddhiAppValidationException("ResourceBatch window's 3rd parameter timeout " +
+                            "should be either int or long, but found " + attributeExpressionExecutors[2]
+                            .getReturnType());
+                }
+            }
+        } else {
+            throw new SiddhiAppValidationException(
+                    "Resource batch window should only have two parameters, " + "but found "
+                            + attributeExpressionExecutors.length + " input attributes");
+        }
+    }
+
+    @Override
+    protected void process(ComplexEventChunk<StreamEvent> streamEventChunk, Processor nextProcessor,
+                           StreamEventCloner streamEventCloner) {
+        synchronized (this) {
+            long currentTime = siddhiAppContext.getTimestampGenerator().currentTime();
+            if (nextEmitTime == -1) {
+                nextEmitTime = currentTime + timeInMilliSeconds;
+                if (scheduler != null) {
+                    scheduler.notifyAt(nextEmitTime);
+                }
+            }
+            ComplexEventChunk<StreamEvent> outputStreamEventChunk = new ComplexEventChunk<StreamEvent>(true);
+            while (streamEventChunk.hasNext()) {
+                StreamEvent streamEvent = streamEventChunk.next();
+                StreamEvent clonedStreamEvent = streamEventCloner.copyStreamEvent(streamEvent);
+                int groupEventMapSize = 0;
+                if (!streamEvent.getType().equals(ComplexEvent.Type.TIMER)) {
+                    Object groupEventMapKey = groupKeyExpressionExecutor.execute(clonedStreamEvent);
+                    if (groupEventMap.containsKey(groupEventMapKey)) {
+                        groupEventMap.get(groupEventMapKey).streamEventList.add(clonedStreamEvent);
```

Comment:
get do groupEventMap.get(groupEventMapKey)
and check if its null and then handle that don't do "containsKey"